configure.ac

```diff
@@ -83,11 +83,18 @@ case $host in
 	backend=haiku
 	platform=posix
 	;;
-wasm32-**)
+*-emscripten)
 	AC_MSG_RESULT([Emscripten])
 	backend=emscripten
 	platform=posix
 	;;
+wasm32*)
+	AC_MSG_ERROR([
+--host=wasm32 used to be recommended and assumed Emscripten backend.
+This is no longer accepted as we might add more WebAssembly backends in the future.
+Please use an explicit --host=wasm32-unknown-emscripten instead.
```

Comment:
I guess we don't need this in a first release.

Reply:
Problem is that there are several projects that I know of that already use this backend from libusb git + potentially more private projects, so I'd rather keep it around for a couple months if you don't mind so that everyone can migrate with graceful error message.

Reply:
> Anyway, I would still prefer leaving out the past tense talk and just say "wasm32 is not accepted, please use...".

Sure, sounds good for future-proofing, I'll update it.

Reply:
Done & also realised that I can suggest the `wasm32-emscripten` shorthand instead of `wasm32-unknown-emscripten` instead, as they're equivalent.